File: dev/golang/atcoder-practice/ABC081B-Shift_only.go
package main

import (
	"fmt"
)

func main() {
	var n, v, cnt int
	min := 1000000000
	fmt.Scan(&n)
	for i := 0; i < n; i++ {
		fmt.Scan(&v)
		cnt = f(0, v)
		if cnt < min {
			min = cnt
		}
	}
	fmt.Println(min)
}

func f(cnt int, i int) int {
	if i%2 == 1 {
		return cnt
	}
	return f(cnt+1, i/2)
}

File: ruby/atcoder-practice/ABC081B-Shift_only.rb
ABC081B-Shift_only.rb
gets.chop.to_i
s = gets.chop.split(' ').map(&:to_i)
@min = 1000000000
def f(cnt, v)
  v % 2 == 1 ? cnt : f(cnt + 1, v / 2)
end
s.each do |n|
  cnt = f(0, n)
  @min = cnt if cnt < @min
end
puts @min
